### PR TITLE
feat: Standardize on using data attributes on `#app` div for "frontend env vars"

### PR DESCRIPTION
**Pull request by jzimbel-mbta**
**Asana task**: ad hoc

This PR does the following:
- adds JS utility functions to easily get values from data attributes on the `#app` div, in much the same way that we use `Application.get_env` on the server
  - this includes a function, `fetchDatasetValue`, that throws an error if a given key is not present, so that we can avoid ending up with `NaN` (from `parseInt(undefined)`) and other bad values that silently cause problems
- uses the new functions throughout the client code, replacing any JS logic that looks at URL query params in the process
- has the server set new data attributes for `is_real_screen` and `screen_side`.

The overall goal is to standardize our approach to passing values from the server to the client, when the values are constant for the lifetime of the client.

Since there are a number of other large PRs that add the new screen simulation logic, I'm happy to wait on merging this until after those higher-priority changes are merged, to avoid slowdowns caused by conflicts.

- [ ] Tests added?

### CODE REVIEW

File: assets/src/util/dataset.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+/**
+ * Gets the entire dataset.
+ */
+export const getDataset = () => document.getElementById("app")?.dataset ?? {};
+
+/**
+ * Gets one value from the dataset, or undefined if the key is not present.
+ */
+export const getDatasetValue = (key: string) => getDataset()[key];
```
Comment (cmaddox5) [inline assets/src/util/dataset.tsx:11]: Just curious, why do we need both `getDatasetValue` and `fetchDatasetValue`? I know one function throws and the other doesn't, but wouldn't we always want to throw?

File: assets/src/hooks/v2/use_audio_readout.tsx [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import { AudioConfig } from "Components/v2/screen_container";
 import useDriftlessInterval from "Hooks/use_driftless_interval";
+import { fetchDatasetValue, getDataset } from "Util/dataset";
```
[the remark this answers is not in the document]
Comment (cmaddox5) [reply assets/src/hooks/v2/use_audio_readout.tsx:3]:
```suggestion
import { fetchDatasetValue } from "Util/dataset";
```

**Review by cmaddox5 — COMMENTED** [review]
Looks great! Surprised we didn't think of this before now. Also really like the new util functions. Just have one question and one suggestion.

File: assets/src/util/dataset.tsx (hunk 1 above)
Comment (jzimbel-mbta) [reply assets/src/util/dataset.tsx:11]: In some cases we're ok with the value not being set, e.g. [the Sentry DSN](https://github.com/mbta/screens/blob/master/lib/screens_web/templates/v2/screen/index.html.eex#L9-L11). `fetch` is for when it has to be there, `get` is for when we expect it to sometimes be undefined. I was going off of Elixir's `Map.get` and `Map.fetch!` functions.

However if you think it makes more sense, I could remove `getDatasetValue` and have the server always set _some_ value (maybe empty string or "none") for every expected data attribute.

Comment (cmaddox5) [reply assets/src/util/dataset.tsx:11]: Ahh I see. Didn't think about those cases. I think two util functions is better than always setting the value in that case. I could imagine that adding new optional attributes could cause errors until we remember that _something_ needs to be there. So I like this.

**Review by cmaddox5 — APPROVED** [review]
Looks great! There is a merge conflict, but I'll go ahead and throw you an approval.

**Pull request merged**